port/ztpf/omrvmem.c

```diff
@@ -0,0 +1,1994 @@
+							break;
+						}
+						/* skip the '-' */
+						next++;
+						if (next >= (lineBuf + lineCursor - 1)) {
+							dataCorrupt = TRUE;
+							break;
+						}
+						end = (uintptr_t) strtoull(next, &next, 16);
+						if ((ULLONG_MAX == end) && (ERANGE == errno)) {
+							dataCorrupt = TRUE;
+							break;
+						}
+
+						currentMmapRange.start = (ADDRESS) start;
+						currentMmapRange.end = (ADDRESS) end;
+						lineCursor = 0;
+
+						if ((currentMmapRange.start >= currentMmapRange.end)
+								|| (currentMmapRange.start < lastMmapRange.end)) {
+							dataCorrupt = TRUE;
+							break;
+						}
+					}
+
+					if (currentMmapRange.start == lastMmapRange.end) {
+						lastMmapRange.end = currentMmapRange.end;
+					} else {
+						/* free block found */
+						AddressRange freeRange;
+						AddressRange intersectAvailable;
+						BOOLEAN haveIntersect = FALSE;
+
+						addressRange_Init(&freeRange, lastMmapRange.end,
+								currentMmapRange.start);
+						memcpy(&lastMmapRange, &currentMmapRange,
+								sizeof(AddressRange));
+
+						/* check if the free block has intersection with the allowed range */
+						haveIntersect = addressRange_Intersect(&allowedRange,
+								&freeRange, &intersectAvailable);
+						if (TRUE == haveIntersect) {
+							uintptr_t intersectSize = addressRange_Width(
+									&intersectAvailable);
+							if (intersectSize >= byteAmount) {
+								memcpy(&lastAvailableRange, &intersectAvailable,
+										sizeof(AddressRange));
+								matchFound = TRUE;
+								if (FALSE == reverse) {
+									break;
+								}
+							}
+						}
+					}
+					lineEnd = FALSE;
+				} /* end proc-line-data */
+			} while (readCursor < bytesRead); /* end proc-chars-loop */
+
+			if ((FALSE == reverse) && (TRUE == matchFound)) {
+				break;
+			}
+
+			if (TRUE == (gotEOF | dataCorrupt)) {
+				break;
+			}
+		} /* end read-file-loop */
+		omrfile_close(portLibrary, fd);
+	} else {
+		dataCorrupt = TRUE;
+	}
+
+	if (TRUE == dataCorrupt) {
+		return NULL;
+	}
+
+	if (TRUE == matchFound) {
+		if (FALSE == reverse) {
+			return lastAvailableRange.start;
+		} else {
+			return (lastAvailableRange.end - byteAmount);
+		}
+	} else {
+		return NULL;
+	}
+}
+
+void
+omrvmem_shutdown(struct OMRPortLibrary *portLibrary)
+{
+#if defined(OMR_PORT_NUMA_SUPPORT)
+	if (NULL != portLibrary->portGlobals) {
+		PPG_numa_platform_supports_numa = 0;
+	}
+#endif
+}
+
+int32_t
+omrvmem_startup(struct OMRPortLibrary *portLibrary)
+{
+	vmem_hugepage_info_t vmem_page_info;
+
+	/* clear page info data, this has the effect of starting off in a standard state */
+	memset(&vmem_page_info, 0x00, sizeof(vmem_hugepage_info_t));
+	get_hugepages_info(portLibrary, &vmem_page_info);
+
+	/* 0 terminate the table */
+	memset(PPG_vmem_pageSize, 0, OMRPORT_VMEM_PAGESIZE_COUNT * sizeof(uintptr_t));
+	memset(PPG_vmem_pageFlags, 0, OMRPORT_VMEM_PAGESIZE_COUNT * sizeof(uintptr_t));
+
+	/* First the default page size */
+#if defined(J9ZTPF)

```

Comment:
What is the build flag used for? Isn't all of the code in this directory ztpf code?

Also would it be possible to rename this flag OMRZPTF?